src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,28 @@
+import { useState } from 'react';
+import { Todo } from '../types/Todo';
+
+export function useLocalStorage(
+  key: string,
+  startValue: Todo[],
+): [Todo[], (v: Todo[]) => void] {
+  const [value, setValue] = useState(() => {
+    const localData = localStorage.getItem(key);
+
+    if (localData === null) {
+      return startValue;
+    }
+
+    try {
+      return JSON.parse(localData) as Todo[];
+    } catch (e) {
+      return startValue;
+    }
+  });
+
+  const save = (newValue: Todo[]) => {
+    localStorage.setItem(key, JSON.stringify(newValue));
```

Comment:
Consider wrapping the `localStorage.setItem` operation in a try/catch to gracefully handle potential errors that can occur from exceeding the storage limit, or in scenarios where the user has disabled the usage of localStorage.

Reply:
Consider following the previous recommendations